File: Js/maximumSubarray.js
console.log(" Maximum Subarrya",getMaximumSubarray([2,3,-6,1,2,-1]))

function getMaximumSubarray(nums) {
    let maxVal = nums[0];
    let current = nums[0];

    for( let i = 1 ; i< nums.length; i++) {
        current = Math.max(nums[i], nums[i]+current);
        maxVal = Math.max(maxVal,current)

    }
    return maxVal;
}

File: Java/MaximumSubarray.java
public class MaximumSubarray {
    public static int maximumSubarray(int[] numbers) {
        int maxValue = numbers[0];
        int current = numbers[0];

        for( int i = 1; i < numbers.length; i ++) {
            current = Math.max(numbers[i], current+numbers[i]);
            maxValue = Math.max(maxValue,current) ;
        }

        return maxValue;
    }
    public static void main(String[] args) {
        int[] numbers= {2,3,-1,1,2,-1};
        System.out.println(" maximum array value  "+maximumSubarray(numbers));

    }
}
